builtin/logical/aws/client.go

```diff
@@ -51,28 +62,38 @@ func getRootConfig(s logical.Storage) (*aws.Config, error) {
 	return &aws.Config{
 		Credentials: creds,
 		Region:      aws.String(credsConfig.Region),
+		Endpoint:    &endpoint,
 		HTTPClient:  cleanhttp.DefaultClient(),
 	}, nil
 }
 
 func clientIAM(s logical.Storage) (*iam.IAM, error) {
-	awsConfig, err := getRootConfig(s)
+	awsConfig, err := getRootConfig(s, "iam")
 	if err != nil {
 		return nil, err
 	}
+
 	client := iam.New(session.New(awsConfig))
+
+	if *awsConfig.Endpoint != "" {
+		client = iam.New(session.New(awsConfig.WithEndpoint(*awsConfig.Endpoint)))
+	}
+
```

Comment:
I don't think this block is necessary anymore, because the endpoint is already in the `awsConfig` object that gets passed in when creating the IAM client.